src/tests/tracing/eventpipe/userevents/userevents.cs

```diff
@@ -0,0 +1,120 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Runtime.InteropServices;
+using System.Threading;
+using Microsoft.Diagnostics.Tracing;
+using Microsoft.Diagnostics.Tracing.Etlx;
+
+namespace Tracing.Tests.UserEvents
+{
+    public class UserEventsTest
+    {
+        private static readonly string trace = "trace.nettrace";
+        private const int SIGINT = 2;
+
+        [DllImport("libc", SetLastError = true)]
+        private static extern int kill(int pid, int sig);
+
+        public static int Main(string[] args)
+        {
+            if (args.Length > 0 && args[0] == "tracee")
+            {
+                UserEventsTracee.Run();
+                return 0;
+            }
+
+            return TestEntryPoint();
+        }
+
+        public static int TestEntryPoint()
+        {
```

Comment:
We should add checks for:
1. process is elevated
2. OS is Linux
3. user_events are supported

Its likely at some point this test will be run in the wrong environment and the logs should make it trivial to diagnose.

Reply:
There is now a UserEventsRequirements helper that checks if the environment meets the requirements.